ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -14033,12 +14033,26 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDedup(BuildCtx & ctx, IHqlExpr
 
     if (!useHash)
     {
-        if (info.compareAllRows)
-            doBuildBoolFunction(instance->classctx, "compareAll", info.compareAllRows);
-        if (!info.keepLeft)
-            doBuildBoolFunction(instance->classctx, "keepLeft", info.keepLeft);
+        // Only necessary to create getFlags() if different from base class getFlags() - i.e. different from default
+        if ( !(!info.compareAllRows && !info.keepBest && info.keepLeft) )
+        {
+            StringBuffer flags;
+            if (info.compareAllRows)
+                flags.append("|DFcompareall");
+            if (info.keepLeft)
+                flags.append("|DFkeepleft");
+            if (info.keepBest)
+                flags.append("|DFkeepbest");
+            if (flags.length()==0)
+                flags.append("|0");
```

Comment:
I think I would prefer to unconditionally build up the flags, and then generate if it didn't match "|DFkeepleft".

Otherwise it is prone to bugs when other options are added.